lib/index.js

```diff
@@ -123,12 +135,18 @@ module.exports = function Agent (serviceToken, environment, options) {
       func.end.call(res, data, encoding)
 
       if (chunked.res.length) {
+        chunked.res = chunked.res.map(function (chunk) {
+          if (chunk instanceof Buffer) {
+            return chunk
+          }
+          return new Buffer(chunk)
+        })
         data = Buffer.concat(chunked.res)
       }
 
       // construct body
-      bodies.res.size = data ? data.length : 0
-      bodies.res.base64 = data ? data.toString('utf8') : ''
+      bodies.res.size = Buffer.byteLength(data)
```

Comment:
isn't this the same as [`.length`](https://nodejs.org/api/buffer.html#buffer_buf_length) unnecessary call here to `.byteLength` since you already have a buffer object.